src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreArguments.cs

```diff
@@ -0,0 +1,210 @@
+        /// </summary>
+#pragma warning disable CA2227 // Collection properties should be read only
+        public Dictionary<string, string> GlobalProperties { get; set; }
+#pragma warning restore CA2227 // Collection properties should be read only
+
+        /// <summary>
+        /// Gets or sets the full path to the MSBuild executable.
+        /// </summary>
+        public string MSBuildExeFilePath { get; set; }
+
+        /// <summary>
+        /// Gets or sets an <see cref="Dictionary{TKey, TValue}" /> containing option names and values.
+        /// </summary>
+#pragma warning disable CA2227 // Collection properties should be read only
+        public Dictionary<string, string> Options { get; set; }
+#pragma warning restore CA2227 // Collection properties should be read only
+
+        /// <summary>
+        /// Reads arguments by searching the specified command-line parameters for an argument file path.
+        /// </summary>
+        /// <param name="args">An array of <see cref="string" /> objects containing the command-line arguments for the current executable.</param>
+        /// <returns>A <see cref="StaticGraphRestoreArguments" /> object if an argument file was found, otherwise <c>null</c>.</returns>
+        public static StaticGraphRestoreArguments Read(string[] args)
+        {
+            // The last command-line argument is the path to the arguments file
+            string argumentFilePath = args.Last();
+
+            if (string.IsNullOrWhiteSpace(argumentFilePath))
+            {
+                return null;
+            }
+
+            FileInfo argumentFile = new FileInfo(argumentFilePath);
+
+            if (!argumentFile.Exists)
+            {
+                return null;
+            }
+
+            return Read(argumentFile.FullName);
+        }
+
+        /// <summary>
+        /// Reads arguments from the specified file path.
+        /// </summary>
+        /// <param name="argumentFilePath">The full path to the argument file to read.</param>
+        /// <returns>A <see cref="StaticGraphRestoreArguments" /> object if the specified argument file was valid, otherwise <c>null</c>.</returns>
+        public static StaticGraphRestoreArguments Read(string argumentFilePath)
+        {
+            using FileStream stream = File.OpenRead(argumentFilePath);
+
+            return Read(stream);
+        }
+
+        /// <summary>
+        /// Writes the current arguments to the specified file.
+        /// </summary>
+        /// <param name="argumentFilePath">The full path to the arguments file to write. if the file exists, it is overwritten. If the directory does not exist, it will be created.</param>
+        public void Write(string argumentFilePath)
+        {
+            var fileInfo = new FileInfo(argumentFilePath);
+
+            fileInfo.Directory.Create();
+
+            using FileStream stream = File.OpenWrite(argumentFilePath);
```

Comment:
If previously existing file is quite large, then overwriting new smaller content into it completely overwrites all previous content or just half of it?

Reply:
Good catch!  [The remarks in the docs for File.OpenWrite](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.openwrite?view=net-6.0#remarks) says:

> If you overwrite a longer string (such as "This is a test of the OpenWrite method") with a shorter string (such as "Second run"), the file will contain a mix of the strings ("Second runtest of the OpenWrite method").

So, we need a `stream.SetLength(stream.Position)` somewhere to ensure that the file is valid.

Reply:
I'll use `File.Create()` instead